cpp/arcticdb/util/error_code.hpp

```diff
@@ -79,6 +79,9 @@ inline std::unordered_map<ErrorCategory, const char*> get_error_category_names()
     ERROR_CODE(5001, E_DUPLICATE_KEY) \
     ERROR_CODE(5002, E_SYMBOL_NOT_FOUND) \
     ERROR_CODE(5003, E_LMDB_MAP_FULL) \
+    ERROR_CODE(5004, E_PERMISSION) \
```

Comment:
Remember to update the documentation for the error codes here: https://docs.arcticdb.io/latest/error_messages/#errors-with-numeric-error-codes